scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -242,6 +243,55 @@
 
   (restore-offers [this pool-name offers])
 
+  (autoscaling? [_]
+    (and
+      (some-> synthetic-pods-config :image count pos?)
+      (some-> synthetic-pods-config :user count pos?)
+      (some-> synthetic-pods-config :max-pods-outstanding pos?)))
+
+  (autoscale! [this pool-name task-requests]
+    (try
+      (assert (cc/autoscaling? this)
+              (str "In " name " compute cluster, request to autoscale despite invalid / missing config"))
+      (let [outstanding-synthetic-pods (->> @all-pods-atom
+                                            (all-pods this)
+                                            (filter controller/synthetic-pod-job-uuid))
+            num-synthetic-pods (count outstanding-synthetic-pods)
+            {:keys [image user command max-pods-outstanding] :or {command "exit 0"}} synthetic-pods-config]
+        (log/info "In" name "compute cluster, there are" num-synthetic-pods
+                  "outstanding synthetic pod(s), and a max of" max-pods-outstanding "are allowed")
+        (if (>= num-synthetic-pods max-pods-outstanding)
+          (log/info "In" name "compute cluster, cannot launch more synthetic pods")
+          (let [using-pools? (config/default-pool)
+                synthetic-task-pool-name (if using-pools? pool-name nil)
+                new-task-requests (remove (fn [{:keys [job]}]
+                                            (some #(= (-> job :job/uuid str) (controller/synthetic-pod-job-uuid %))
+                                                  outstanding-synthetic-pods))
+                                          task-requests)
+                task-metadata-seq (->>
+                                    new-task-requests
+                                    (map (fn [{:keys [job]}]
+                                           {:task-id (str (UUID/randomUUID))
```

Comment:
Don't use UUID/randomUUID

Reply:
Nitpick: In addition, we should use a UUID generator that has a well-known prefix/suffix that makes it easy to identify synthetic pods.

Reply:
Good idea, will do

Reply:
Use d/squuid, as is used for the other tasks.

Reply:
Done, we're now using `synthetic-{pool}-{d/squuid}`